### PR TITLE
 [ServiceBus] Emplacing TestRegionOverride into tests.yml to enable canary testability

### PR DESCRIPTION
**Pull request by KieranBrantnerMagee**
Initial attempt at plumbing TestRegionOverride through the tests.yml configurations to allow for canary testability.

-----
Context:
- The service team has canary test in certain areas
- We want to test against canary to make sure everything still tests OK/creates/tearsdown
  - regression
- This might need a new pipeline which runs weekly/nightly (TBD)
- Java has enabled this, check with Java team or Ramya for more.

Action item:
- local test first
- reach out to java/ramya to see what are the next steps

### CODE REVIEW

File: eng/pipelines/templates/jobs/archetype-sdk-tests.yml [hunk 1]
```diff
@@ -69,6 +69,7 @@ jobs:
           CoverageArg: $(CoverageArg)
           EnvVars:
             AZURE_RUN_MODE: 'Live' #Record, Playback
+            RESOURCE_REGION: $(TestRegionOverride)
```
Comment (scbedd) [inline eng/pipelines/templates/jobs/archetype-sdk-tests.yml:72]: After you add the parameter at the top. Swap to use `#{{ parameters.RegionOverride }}`

**Comment by mitchdenny** [comment]
Does this PR get superseded by the work that @benbp has been doing?

**Comment by benbp** [comment]
@KieranBrantnerMagee you can add a parameter like this to test canary:

```
parameters:
  Clouds: 'Public,Canary'
```

Currently it defaults to centraluseuap as you have set. If you want to customize location per cloud:

```
parameters:
  Clouds: 'Public,Canary'
  CloudConfig:
      Public:
        SubscriptionConfiguration: $(sub-config-azure-cloud-test-resources)
      Canary:
        SubscriptionConfiguration: $(sub-config-azure-cloud-test-resources)
        Location: 'eastus2euap'
```

File: sdk/servicebus/tests.yml [hunk 2]
```diff
@@ -8,11 +8,20 @@ stages:
       DeployArmTemplate: true
       TestSamples: true
       BuildTargetingString: azure-servicebus*
+      TestCanary: false
```
Comment (yunhaoling) [inline sdk/servicebus/tests.yml:11]:
```suggestion
```

**Review by benbp — APPROVED** [review]

File: sdk/servicebus/tests.yml [hunk 3]
```diff
@@ -14,5 +14,14 @@ stages:
...
         AZURE_CLIENT_SECRET: $(aad-azure-sdk-test-client-secret)
         AZURE_TEST_RUN_LIVE: 'true'
-      MatrixFilters:
-        - '"PythonVersion=^(?!pypy3).*"'
+      Clouds: 'Public,Canary'
+      CloudConfig:
+        Public:
+          SubscriptionConfiguration: $(sub-config-azure-cloud-test-resources)
+          MatrixFilters:
+            - PythonVersion=^(?!pypy3).*
```
[the remark this answers is not in the document]
Comment (yunhaoling) [reply sdk/servicebus/tests.yml:22]: thanks for your suggestion @benbp , matrix updated!

**Pull request merged**